src/javascript/tests.rs

```diff
@@ -181,3 +181,30 @@ const e = [
 "#
     );
 }
+
+#[test]
+fn external_functions() {
+    assert_js!(
+        r#"
+pub external type Thing
+external fn foo() -> Thing = "document" "foo"
+pub external fn bar(Int, String) -> Thing = "document" "bar"
+external fn baz(x: Int, y: String) -> Thing = "document" "baz"
+ 
+"#,
+        r#""use strict";
+
+function foo() {
+  return document.foo()
```

Comment:
Oh, I just had a thought. We're probably going to have to `import { foo } from "document";` ?

Reply:
No, don't think so. My example project I had working in the browser didn't require any of that.
I guess bundlers might need separate thinking about, but in those cases it's probably a setting in the bundler to flag what globals are referenced.

Reply:
Let's do this in a later PR

Reply:
Actually I just had an idea how how externals could work that may work well and means we don't need to do this